go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -282,18 +293,12 @@ func (c *Controller) Run(stopCh <-chan struct{}, wg *sync.WaitGroup, threads int
 				return fmt.Errorf("failed to create rule in chain %s to save pkt marking: %v", utiliptables.ChainPrerouting, err)
 			}
 		}
-	}
-	if ovnconfig.Gateway.Mode == ovnconfig.GatewayModeLocal {
+
 		// If dst is a node IP, use main routing table and skip EIP routing tables
-		if err = c.ruleManager.Add(getNodeIPFwMarkIPRule()); err != nil {
+		// src_valid_mark is not applicable to ipv6
+		if err = c.ruleManager.Add(getNodeIPFwMarkIPRule(netlink.FAMILY_V6)); err != nil {
```

Comment:
why isnt this inside the condition `ovnconfig.Gateway.Mode == ovnconfig.GatewayModeLocal  {`

Reply:
It should, the github diff bamboozled me.
Thanks!